cmdstanpy/lib.py

```diff
@@ -527,27 +552,59 @@ def sample(self) -> np.ndarray:
         all draws from a chain are contiguous.
         """
         if self._sample is None:
-            self._sample = self.get_sample()
+            self.parse_sample()
         return self._sample
 
-    def get_sample(self) -> np.ndarray:
+    def parse_sample(self) -> None:
         """
-        Returns posterior sample.
-        The first time this function is called it assembles the sample
-        from the stan_csv files; subsequent calls to this function
-        return the assembled sample.
+        Allocates and populates the stepsize, metric, and drawset arrays
+        by parsing the validated stan_csv files if they don't already exist.
         """
-        sample = np.empty(
+        if not (self._stepsize is None and self._metric is None and
+                    self._sample is None):
+            return
```

Comment:
This could have explicit `None` but can be without. I guess it depends on the linter what is recommend.